src/NuGet.Clients/VisualStudio/RestoreManager/IVsProjectRestoreInfo.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// Contains complete project details needed for restore operation
+    /// </summary>
+    [ComImport]
+    [Guid("ab43992d-b977-436d-84c2-e76aeed20de2")]
+    public interface IVsProjectRestoreInfo
+    {
+        /// <summary>
+        /// Friendly project name.
+        /// </summary>
+        string ProjectName { get; }
+
+        /// <summary>
+        /// Name unique to the project across the solution.
+        /// </summary>
+        string ProjectUniqueName { get; }
+
+        /// <summary>
+        /// MSBuild project file path.
+        /// </summary>
+        string ProjectPath { get; }
+
+        /// <summary>
+        /// Assets file output path.
+        /// </summary>
+        string OutputPath { get; }
+
+        /// <summary>
+        /// User packages folder path.
+        /// </summary>
+        string PackagesPath { get; }
```

Comment:
Need to be more specific, I can't tell if this is the packages.config folder or the global one scoped.

What is the purpose of this interface? Producer/consumer?